Zotero/Scenes/Detail/ItemDetail/Views/ItemDetailViewController.swift

```diff
@@ -53,36 +69,77 @@ final class ItemDetailViewController: UIViewController {
+                self?.update(to: state)
+            })
+            .disposed(by: disposeBag)
+
+        func setupFileObservers() {
+            NotificationCenter.default.rx
+                .notification(.attachmentFileDeleted)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak viewModel] notification in
+                    guard let viewModel, let notification = notification.object as? AttachmentFileDeletedNotification else { return }
+                    viewModel.process(action: .updateAttachments(notification))
+                })
+                .disposed(by: disposeBag)
+
+            NotificationCenter.default.rx
+                .notification(UIApplication.willEnterForegroundNotification)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] _ in
+                    guard let self else { return }
+                    // Need to reload data to current state before going back to foreground. iOS reloads the collection view when going to foreground with current snapshot. When
+                    // editing text fields we don't update the snapshot (so that the cell is not reloaded while typing), so the edited fields are reset to previous state.
+                    collectionViewHandler.reloadAll(to: viewModel.state, animated: false)
+                })
+                .disposed(by: disposeBag)
+
+            controllers.userControllers?.fileDownloader.observable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] update in
+                    guard let self else { return }
+                    viewModel.process(action: .updateDownload(update))
+
+                    guard viewModel.state.attachmentToOpen == update.key else { return }
+
+                    // 🍎 check logic of processes
+                    switch update.kind {
+                    case .ready:
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachment(key: update.key, parentKey: update.parentKey, libraryId: update.libraryId)
+
+                    case .failed(let error):
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachmentError(error)
+
+                    case .cancelled:
+                        viewModel.process(action: .attachmentOpened(update.key))
+
+                    case .progress:
+                        return
+                    }
+                })
+                .disposed(by: self.disposeBag)
+        }
     }
 
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        self.didAppear = true
+    override func viewIsAppearing(_ animated: Bool) {
+        super.viewIsAppearing(animated)
+        viewModel.process(action: .loadInitialData)
```

Comment:
Any difference having it here rather than in `viewDidLoad`?

Reply:
No difference user-wise, it just simplifies the code by remving all needs for a `didAppear` varible.

Reply:
Yes, my bad, I'll fix it.

Reply:
Check https://github.com/zotero/zotero-ios/pull/1021/commits/4379c399244ba73fb0770afda77bc7d95f14e28b